src/hyperv/hyperv_api_v1.c

```diff
@@ -1594,20 +1594,28 @@ hyperv1DomainAttachSyntheticEthernetAdapter(virDomainPtr domain,
     char *connection__PATH = NULL;
     unsigned char guid[VIR_UUID_BUFLEN];
     virBuffer query = VIR_BUFFER_INITIALIZER;
-    Msvm_ComputerSystem_V1 *computer = NULL;
+    Msvm_VirtualSwitch_V1 *vSwitch = NULL;
     eprParam virtualSwitch_REF, ComputerSystem_REF;
     simpleParam virtualSwitch_Name, virtualSwitch_FriendlyName,
                 virtualSwitch_ScopeOfResidence;
     embeddedParam ResourceSettingData;
     invokeXmlParam *params = NULL;
     properties_t *NewResources = NULL;
 
+    /* get the vSwitch first so we can get data from it as needed */
```

Comment:
Why is this necessary? The only thing you need from this is the GUID of the Msvm_VirtualSwitch object, which is already present in the XML (`net->data.bridge.brname`). You even use it in the error message on line 1658.